### ISSUE TITLE
.NET 10: PlatformCompatibilityAnalyzer not working

### ISSUE THREAD

**Issue by jonathanpeppers** [dotnet/roslyn-analyzers#7525]
### Analyzer

**Diagnostic ID**: [CA1416](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1416): `Validate platform compatibility`

### Analyzer source

* https://github.com/dotnet/roslyn-analyzers/blob/5ed336762c6260a83ece35cd1f6749251452bad0/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

### Describe the bug

For Android, we have an API such as:
```csharp
[SupportedOSPlatform("android22.0")]
public virtual int AccessibilityTraversalAfter { get; set; }
```

Using the API with `SupportedOSPlatformVersion=21.0` does not result in a `CA1416` warning.

We first noticed this here, getting the Android workload on a .NET 10 SDK:

* https://github.com/dotnet/android/pull/9395

In the following integration test:

* https://github.com/dotnet/android/blob/208c52986a9369b0dab196eb6ab0618b521e2f98/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/ManifestTest.cs#L1071

### Steps To Reproduce

* `dotnet new android`
* Use `button.AccessibilityTraversalAfter`
* Observe no warnings

Here is a `.binlog` if that is helpful: [supportedosplatformversion.zip](https://github.com/user-attachments/files/18364927/supportedosplatformversion.zip)

### Expected behavior

Using the API with `SupportedOSPlatformVersion=21.0` results in a `CA1416` warning.

### Actual behavior

Using the API with `SupportedOSPlatformVersion=21.0` does *not* result in a `CA1416` warning.

## Additional context

I'm guessing the code here would not work as expected for `TargetFramework=net10.0`, as it has 1 more digit:

https://github.com/dotnet/roslyn-analyzers/blob/5ed336762c6260a83ece35cd1f6749251452bad0/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs#L284-L289

**Comment by ViktorHofer** [comment]
Yes the code that you shared looks faulty. Just wanted to add that a component should never parse and interpret the `TargetFramework` property. Instead, the deferred properties: i.e. TFI and TFV should be read.

Why? Because with the eventual TFM aliasing feature, the analyzer will stop working:

```xml
<PropertyGroup>
  <TargetFrameworks>vs17.12;vs17.13</TargetFrameworks>
</PropertyGroup>

<PropertyGroup>
  <TargetFrameworkIdentifier>.NETCoreApp</TargetFrameworkIdentifier>
  <TargetFrameworkVersion>10.0</TargetFrameworkVersion>
</PropertyGroup>

<PropertyGroup>
  <RoslynAnalyzerVersion Condition="'$(TargetFramework)' == 'vs17.12'">4.12</RoslynAnalyzerVersion>
  <RoslynAnalyzerVersion Condition="'$(TargetFramework)' == 'vs17.13'">4.13</RoslynAnalyzerVersion>
</PropertyGroup>
```

**Comment by ViktorHofer** [comment]
@dotnet/dotnet-analyzers @jeffhandley please acknowledge the issue. This should really get fixed before we ship P1. Code complete is in 10 days.
